From 0d92c557a94d867915c4e68d70090c87961043e7 Mon Sep 17 00:00:00 2001
From: Haoran Wang <b50027@freescale.com>
Date: Tue, 20 Oct 2015 16:26:07 +0800
Subject: [PATCH 549/602] MA-6450-3 Enable BCM4339 on sabresd_6sx

Modify init.rc and private lib for wifi
and bluetooth on sabresd_6sx.

Signed-off-by: Haoran Wang <b50027@freescale.com>
---
 imx6/etc/ueventd.freescale.rc |  6 +-----
 sabresd_6sx/BoardConfig.mk    | 13 ++++++++++++-
 sabresd_6sx/init.rc           | 12 ++++++++++++
 3 files changed, 25 insertions(+), 6 deletions(-)

diff --git a/imx6/etc/ueventd.freescale.rc b/imx6/etc/ueventd.freescale.rc
index a666615..db57bd9 100644
--- a/imx6/etc/ueventd.freescale.rc
+++ b/imx6/etc/ueventd.freescale.rc
@@ -1,4 +1,5 @@
 /dev/pmem_gpu             0660   system     graphics
+/dev/ttymxc2              0660   bluetooth  bluetooth
 /dev/ttymxc4              0660   bluetooth  bluetooth
 /dev/snd/*                0660   system     audio
 /dev/ttyUSB*              0640   radio      radio
@@ -11,11 +12,6 @@
 /dev/mxc_ipu              0660   media      system
 /dev/mxc_vpu              0660   media      system
 /dev/uinput               0660   system     bluetooth
-/dev/ttymxc2              0660   system     gps
-/dev/ttya0                0660   system     gps
-/dev/ptya0                0660   system     gps
-/dev/ttya1                0660   system     gps
-/dev/ptya1                0660   system     gps
 /dev/ion                  0666   media      system
 /dev/galcore	          0666 	 system     graphics
 /dev/graphics/fb0         0660   system     graphics
diff --git a/sabresd_6sx/BoardConfig.mk b/sabresd_6sx/BoardConfig.mk
index 453a6b2..7dff636 100644
--- a/sabresd_6sx/BoardConfig.mk
+++ b/sabresd_6sx/BoardConfig.mk
@@ -41,12 +41,16 @@ BOARD_WPA_SUPPLICANT_DRIVER  := NL80211
 BOARD_HOSTAPD_DRIVER         := NL80211
 TARGET_KERNEL_MODULES        := \
                                 kernel_imx/net/wireless/cfg80211.ko:system/lib/modules/cfg80211_realtek.ko \
-                                kernel_imx/drivers/net/wireless/rtlwifi/rtl8821as/8821as.ko:system/lib/modules/8821as.ko
+                                kernel_imx/drivers/net/wireless/rtlwifi/rtl8821as/8821as.ko:system/lib/modules/8821as.ko \
+                                kernel_imx/drivers/net/wireless/bcmdhd/bcmdhd.ko:system/lib/modules/bcmdhd.ko
 
 BOARD_HOSTAPD_PRIVATE_LIB_QCOM              := lib_driver_cmd_qcwcn
 BOARD_WPA_SUPPLICANT_PRIVATE_LIB_QCOM       := lib_driver_cmd_qcwcn
 BOARD_HOSTAPD_PRIVATE_LIB_RTL               := lib_driver_cmd_rtl
 BOARD_WPA_SUPPLICANT_PRIVATE_LIB_RTL        := lib_driver_cmd_rtl
+BOARD_HOSTAPD_PRIVATE_LIB_BCM               := lib_driver_cmd_bcmdhd
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB_BCM        := lib_driver_cmd_bcmdhd
+
 #for intel vendor
 ifeq ($(BOARD_WLAN_VENDOR),INTEL)
 BOARD_HOSTAPD_PRIVATE_LIB                := private_lib_driver_cmd
@@ -59,6 +63,11 @@ WIFI_DRIVER_MODULE_NAME                  := "iwlagn"
 WIFI_DRIVER_MODULE_PATH                  ?= auto
 endif
 
+WIFI_DRIVER_FW_PATH_STA        := "/system/etc/firmware/bcm/fw_bcmdhd.bin"
+WIFI_DRIVER_FW_PATH_P2P        := "/system/etc/firmware/bcm/fw_bcmdhd.bin"
+WIFI_DRIVER_FW_PATH_AP         := "/system/etc/firmware/bcm/fw_bcmdhd_apsta.bin"
+WIFI_DRIVER_FW_PATH_PARAM      := "/sys/module/bcmdhd/parameters/firmware_path"
+
 BOARD_MODEM_VENDOR := AMAZON
 
 USE_ATHR_GPS_HARDWARE := true
@@ -100,6 +109,8 @@ endif
 
 # atheros 3k BT
 BOARD_USE_AR3K_BLUETOOTH := true
+
+BOARD_HAVE_BLUETOOTH_BCM := true
 BOARD_BLUETOOTH_BDROID_BUILDCFG_INCLUDE_DIR := device/fsl/sabresd_6sx/bluetooth
 
 USE_ION_ALLOCATOR := false
diff --git a/sabresd_6sx/init.rc b/sabresd_6sx/init.rc
index b3c8cd2..7b6fe6a 100755
--- a/sabresd_6sx/init.rc
+++ b/sabresd_6sx/init.rc
@@ -222,6 +222,18 @@ service rtw_supplc_adv /system/bin/rtl_wpa_supplicant \
     disabled
     oneshot
 
+service bcm_supplicant /system/bin/bcm_supplicant \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
+    -I/system/etc/wifi/wpa_supplicant_overlay.conf \
+    -puse_p2p_group_interface=1p2p_device=1 \
+    -g/data/misc/wifi/sockets/p2p0 \
+    -O/data/misc/wifi/sockets \
+    -e/data/misc/wifi/entropy.bin -g@android:wpa_wlan0
+    socket wpa_wlan0 dgram 660 wifi wifi
+    class late_start
+    disabled
+    oneshot
+
 service wpa_supplicant /system/bin/wpa_supplicant \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
     -I/system/etc/wifi/wpa_supplicant_overlay.conf \
-- 
1.8.0

